lib/new-admin/graphql/types/users.type.js

```diff
@@ -0,0 +1,68 @@
+const typeDef = `
+  type UserSession {
+    sid: String!
+    sess: JSONObject!
+    expire: Date!
+  }
+
+  type User {
+    id: ID
+    username: String
+    role: String
+    enabled: Boolean
+    created: Date
+    last_accessed: Date
+    last_accessed_from: String
+    last_accessed_address: String
+  }
+
+  type TwoFactorSecret {
+    user_id: ID
+    secret: String!
+    otpauth: String!
+  }
+
+  type ResetToken {
+    token: String
+    user_id: ID
+    expire: Date
+  }
+
+  type RegistrationToken {
+    token: String
+    username: String
+    role: String
+    expire: Date
+  }
+
+  type Query {
+    users: [User] @auth(requires: [SUPERUSER])
+    sessions: [UserSession] @auth(requires: [SUPERUSER])
+    userSessions(username: String!): [UserSession] @auth(requires: [SUPERUSER])
+    userData: User
+    get2FASecret(username: String!, password: String!): TwoFactorSecret
+    confirm2FA(code: String!): Boolean @auth(requires: [SUPERUSER])
+    validateRegisterLink(token: String!): User
+    validateResetPasswordLink(token: String!): User
+    validateReset2FALink(token: String!): TwoFactorSecret
+  }
+
+  type Mutation {
+    deleteUser(id: ID!): User @auth(requires: [SUPERUSER])
+    deleteSession(sid: String!): UserSession @auth(requires: [SUPERUSER])
+    deleteUserSessions(username: String!): [UserSession] @auth(requires: [SUPERUSER])
+    changeUserRole(id: ID!, newRole: String!): User @auth(requires: [SUPERUSER])
+    toggleUserEnable(id: ID!): User @auth(requires: [SUPERUSER])
+    login(username: String!, password: String!): String
+    input2FA(username: String!, password: String!, code: String!, rememberMe: Boolean!): Boolean
+    setup2FA(username: String!, password: String!, secret: String!, codeConfirmation: String!): Boolean
+    createResetPasswordToken(userID: ID!): ResetToken @auth(requires: [SUPERUSER])
+    createReset2FAToken(userID: ID!): ResetToken @auth(requires: [SUPERUSER])
+    createRegisterToken(username: String!, role: String!): RegistrationToken @auth(requires: [SUPERUSER])
+    register(username: String!, password: String!, role: String!): Boolean
+    resetPassword(userID: ID!, newPassword: String!): Boolean
+    reset2FA(userID: ID!, secret: String!, code: String!): Boolean
```

Comment:
Dangerous because of the lack of auth requirement. They need to come with a token to validate